### PR TITLE
Simple upgrade test using kubetest2 framework

### PR DESCRIPTION
**Pull request by justinsb**
Starting very simple and hard-coded!

### CODE REVIEW

**Comment by justinsb** [comment]
cc @rifelpet I think this is the right way to do things?

/assign rifelpet

File: tests/e2e/scenarios/upgrade/run-test [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+trap finish EXIT
+
+kubetest2 kops ${KUBETEST2_COMMON_ARGS} --up
+
+KUBECONFIG=${HOME}/.kube/config
+TEST_ARGS="--kubeconfig=${KUBECONFIG}"
+if [[ "${CLOUD_PROVIDER}" == "aws" ]]; then
+  ZONES=`kops get cluster ${CLUSTER_NAME} -ojson | jq -r .spec.subnets[].zone`
+  CLUSTER_TAG="${CLUSTER_NAME}"
+  TEST_ARGS="${TEST_ARGS} --provider=aws --gce-zone=${ZONES[0]} --cluster-tag=${CLUSTER_TAG}"
```
Comment (rifelpet) [inline tests/e2e/scenarios/upgrade/run-test:55]: Might be worthy of a comment that --gce-zone is actually provider agnostic (this tripped me up for a minute)

https://github.com/kubernetes/test-infra/blob/1a5044709fb8b5350c99a2a8e5ac208827a04c07/kubetest/kops.go#L692-L697

File: tests/e2e/scenarios/upgrade/run-test [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+kops set cluster ${CLUSTER_NAME} cluster.spec.kubernetesVersion=1.20.0
+kops update cluster
+kops update cluster --admin --yes
+
+kops rolling-update cluster
+kops rolling-update cluster --yes
+
+kops validate cluster
+
+kubetest2 kops ${KUBETEST2_COMMON_ARGS} --down
```
Comment (rifelpet) [inline tests/e2e/scenarios/upgrade/run-test:82]: any reason not to just let the trap function run every time? You could remove these last 3 lines and simplify the two functions above.

Comment (justinsb) [reply tests/e2e/scenarios/upgrade/run-test:82]: My thought was that we wanted to preserve the error, but actually I think that you're right and it's just over-complicating things - we want to know if down failed and there's still a lingering cluster!

File: tests/e2e/scenarios/upgrade/run-test (hunk 1 above)
Comment (justinsb) [reply tests/e2e/scenarios/upgrade/run-test]: Good call - added

File: tests/e2e/scenarios/upgrade/run-test [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+# Always tear-down the cluster when we're done
+function kubetest2_down {
+  kubetest2 kops ${KUBETEST2_COMMON_ARGS} --down
+}
+function finish {
+  if [[ "${RUN_KUBETEST2_DOWN}" == "y" ]]; then
+    kubetest2_down || echo "kubetest2 down failed"
+  fi
+}
+trap finish EXIT
```
Comment (rifelpet) [inline tests/e2e/scenarios/upgrade/run-test:46]:
```suggestion
function finish {
  kubetest2 kops ${KUBETEST2_COMMON_ARGS} --down || echo "kubetest2 down failed"
}
trap finish EXIT
```

**Comment by justinsb** [comment]
Thanks @rifelpet - I incorporated your change!

**Comment by rifelpet** [comment]
👍🏻 
/lgtm

**Pull request merged**